pysnn/neuron.py

```diff
@@ -299,8 +299,99 @@ def __init__(
     def update_trace(self, x):
         spikes = self.convert_spikes(x)
         self.trace = self.trace_update(
-            self.trace, spikes, self.alpha_t, self.trace_decay, self.dt
+            self.trace, spikes, self.alpha_t, self.tau_t, self.dt
+        )
+
+    def update_voltage(self, x):
+        self.v_cell = self.voltage_update(
+            self.v_cell,
+            self.v_rest,
+            x,
+            self.alpha_v,
+            self.tau_v,
+            self.dt,
+            self.refrac_counts,
+        )
+
+    def forward(self, x):
+        x = self.fold(x)
+        self.update_voltage(x)
+        spikes = self.spiking()
+        self.update_trace(spikes)
+        self.refrac(spikes)
+        if self.complete_trace is not None:
+            self.concat_trace(spikes)
+        return spikes, self.trace
+
+
+#########################################################
+# Adaptive LIF Neuron
+#########################################################
+class AdaptiveLIFNeuron(BaseNeuron):
+    r"""Adaptive Leaky Integrate and Fire neuron."""
+
+    def __init__(
+        self,
+        cells_shape,
+        thresh,
+        v_rest,
+        alpha_v,
+        alpha_t,
+        dt,
+        duration_refrac,  # From here on class specific params
+        tau_v,
+        tau_t,
+        alpha_thresh,
+        tau_thresh,
+        update_type="linear",
+        store_trace=False,
+    ):
+        super(AdaptiveLIFNeuron, self).__init__(
+            cells_shape,
+            thresh,
+            v_rest,
+            alpha_v,
+            alpha_t,
+            dt,
+            duration_refrac,
+            store_trace=store_trace,
+        )
+
+        # TODO: here or in separate function? Or overwrite init_neuron?
+        # Type of updates
+        if update_type == "linear":
+            self.voltage_update = sf._lif_linear_voltage_update
+            self.trace_update = sf._linear_trace_update
+            self.thresh_update = sf._linear_thresh_update
+        elif update_type == "exponential":
+            self.voltage_update = sf._lif_voltage_update
```

Comment:
Update sf._lif_voltage_update to sf._lif_exponential_voltage_update to be in line with other names.